provisioning/containers/gui-common/websockify/websockify.go

```diff
@@ -86,23 +84,28 @@ func (h *NoVncHandler) serveWs(w http.ResponseWriter, r *http.Request) {
 	}
 
 	ip := r.Header.Get(headerXForwardedFor)
-	connID := atomic.AddUint32(&h.connectionsCount, 1)
+	if ip == "" {
+		ip = "unknown"
+	}
+	connUID := r.URL.Query().Get("connUid")
 
-	metric := makeLatencyObserver(ip, fmt.Sprint(connID))
+	connectionInfo := &ConnInfo{IP: ip, UID: connUID}
```

Comment:
I don't get it. When this `connectionInfo` object is created, it is still not a value of the `connectionsTracking sync.Map`. It will be inserted as value mapped to `connUID` after the first ping-pong cycle.
I'm sorry if I misunderstood your message